src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,152 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+
+    private int size;
+
+    private Node<K, V>[] bucketArray;
+
+    public MyHashMap() {
+        bucketArray = new Node[INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
+        if (key == null) {
+            handleNullKey(value);
+            return;
+        }
 
+        int bucketIndex = getBucketIndex(key);
+        if (bucketArray[bucketIndex] == null) {
+            bucketArray[bucketIndex] = new Node<>(key, value);
+        } else {
+            Node<K, V> current = bucketArray[bucketIndex];
+            while (current != null) {
+                if (key.equals(current.key)) {
```

Comment:
Just perform null-safe comparison and you're fine
```suggestion
                if (key == current.key || key != null && key.equals(current.key)) {
```